services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/query/MainQuery.java

```diff
@@ -0,0 +1,77 @@
+package org.sagebionetworks.repo.manager.table.query;
+
+import org.sagebionetworks.repo.model.table.Query;
+import org.sagebionetworks.table.cluster.CombinedQuery;
+import org.sagebionetworks.table.cluster.SqlQuery;
+import org.sagebionetworks.table.cluster.SqlQueryBuilder;
+import org.sagebionetworks.table.cluster.TranslationDependencies;
+
+public class MainQuery {
+
+	private final SqlQuery sqlQuery;
+
+	private MainQuery(String startingSql, TranslationDependencies dependencies, Query query, Long maxBytesPerPage) {
+
+		CombinedQuery combined = CombinedQuery.builder().setQuery(startingSql)
+				.setSchemaProvider(dependencies.getSchemaProvider()).setOverrideOffset(query.getOffset())
+				.setOverrideLimit(query.getLimit()).setSelectedFacets(query.getSelectedFacets())
+				.setSortList(query.getSort()).setAdditionalFilters(query.getAdditionalFilters()).build();
+
+		sqlQuery = new SqlQueryBuilder(combined.getCombinedSql(), dependencies).maxBytesPerPage(maxBytesPerPage)
```

Comment:
SqlQuery should really be renamed to SynapseSQLTRanslator or something. The builder should be part of the class itself.